hera_cal/lstbin.py

```diff
@@ -601,39 +673,45 @@ def lst_bin_files(data_files, input_cals=None, dlst=None, verbose=True, ntimes_p
     kwargs['start_jd'] = start_jd
     integration_time = np.median(hd.integration_time)
     assert np.all(np.abs(np.diff(times) - np.median(np.diff(times))) < 1e-6), 'All integrations must be of equal length (BDA not supported).'
-
-    # get baselines from data and form baseline groups
-    bls = sorted(hd.get_antpairs())
-    Nbls = len(bls)
+    # get antpos over all nights
+    for dlist in data_files:
+        hd = io.HERAData(dlist[-1])
+        for a in hd.antpos:
+            if a not in antpos:
+                antpos[a] = hd.antpos[a]
+    # generate a list of dictionaries which contain the nights occupied by each unique baseline
+    # (or unique baseline group if average_redundant_baselines is true)
+    bldicts = gen_bldicts([io.HERAData(dlists[-1]) for dlists in data_files], bl_error_tol=bl_error_tol, include_autos=include_autos,
```

Comment:
can we rename this variable and this function. `bldict` is so generic and this is a new and very specialized object. The code would be more readable if it were something like `bl_nightly_dict` or maybe even more specific than that...

Reply:
I've switched to `bl_nightly_dict`.